src/main.rs

```diff
@@ -161,6 +176,12 @@ fn main() {
             }
         }
         ("explain", Some(sub_m)) => print::explain(sub_m.clone()),
-        _ => {}
+        ("completions", Some(sub_m)) => {
+            let shell_name = sub_m.value_of("shell").expect("Shell name missing.");
+            let shell = Shell::from_str(shell_name).expect("Invalid shell");
```

Comment:
Is it safe to let clap generate completions for an unknown shell? (I tried looking this up, but I can't seem to find a quick answer)

Reply:
Clap won't generate completions for an unknown shell. The valid shells are in the [`clap::Shell`](https://docs.rs/clap/2.33.0/clap/enum.Shell.html) enum; anything else will trigger an error.